### ISSUE TITLE
Number of discriminator output

### ISSUE THREAD

**Issue by shahriarshayesteh** [crux82/ganbert-pytorch#1]
Hello,
Thanks for sharing your code. I am trying to understand the GAN-BERT paper. I faced an issue in the code that the reason is not clear to me.

In the python notebook, we have a dataset with 50 classes, in the **"label_list"** the name of each class is given, also **"UNK_UNK"** is added for unlabeled data. Therefore wherever **len(label_list)**  is used, it is equal to 51. 

On the other side, the discriminator's output is the number of classes +1, since discriminator is not only should discriminate between real and the fake examples, also need to classify real ones. Therefore, if an example is fake, it should be classified as class 51, otherwise, the discriminator assigns a class to the example. 

Here is what I do not understand, when we are going to initialize the discriminator, we use**len(label_list)** = 51, as the input argument for the number of classes. inside the discriminator also add +1 to the 51, so the discriminator output is 52.

Then, when the supervised learning loss is being calculated in the training phase, we take the **logits = D_real_logits[:,0:-1]** which means all the output logit except the last one (which is related to the example being fake or real and the last logit is used to calculate the unsupervised loss). In this code, **logits = D_real_logits[:,0:-1]** length is 51 while we have 50 classes. Also, in the evaluation time, when the test prediction and loss on the test set are calculated, **filtered_logits** takes all the output logit except the last one has the length of 51 while we have 50 labels. I wondered if there is any problem with the code or I did not fully understand the paper?

Thanks!

**Comment by crux82** [comment]
As in the original GANBERT, the additional UNK_UNK class is only used to handle the labels assigned to .... unlabeled data ... from the input file. The output vector is augmented with an additional label just to simplify the data loading phase. 
As you pointed out, it adds an additional (and useless) dimension that should not be confused with the one added to Adversarial learning. 

Hope it helps 

Danilo
